pkg/sfu/peer.go

```diff
@@ -24,28 +24,36 @@ var (
 	ErrOfferIgnored = errors.New("offered ignored")
 )
 
+type Peer interface {
+	ID() string
+	Session() Session
+	Publisher() *Publisher
+	Subscriber() *Subscriber
+	Close() error
+}
+
 // JoinConfig allow adding more control to the peers joining a session.
 type JoinConfig struct {
 	// If true the peer will not be allowed to publish tracks to session.
 	NoPublish bool
 	// If true the peer will not be allowed to subscribe to other peers in session.
 	NoSubscribe bool
-	// If true it will relay all the published tracks of the peer
+	// If true it will Relay all the published tracks of the peer
 	Relay bool
 }
 
-// SessionProvider provides the session to the sfu.Peer{}
+// SessionProvider provides the session to the sfu.Peer
 // This allows the sfu.SFU{} implementation to be customized / wrapped by another package
 type SessionProvider interface {
-	GetSession(sid string) (*Session, WebRTCTransportConfig)
+	GetSession(sid string) (*session, WebRTCTransportConfig)
```

Comment:
Making session private there breaks the purpose of this interface.  I don't quite understand how this is meant to be used.  Shouldn't this be using the Session interface?